From 48d596c41b17ccd9672c370ce2586fd3f8f2e996 Mon Sep 17 00:00:00 2001
From: Daniel Baluta <daniel.baluta@nxp.com>
Date: Tue, 27 Nov 2018 14:58:45 +0200
Subject: [PATCH 5223/5242] MLK-20095-2: arm64: dts: Make pd_asrc0 dependent
 on pd_esai0

commit  7c5c84b7fea929f3a483100195939378d6d0e48d from
https://source.codeaurora.org/external/imx/linux-imx.git

Because now DSP supports ASRC we need to power up pd_asrc0
each time pd_esai0 is powered up.

Current power domain tree looks like this:

	* pd_audio
	  * pd_audio_clk0
	    * pd_audio_clk1
	      * pd_dma0_chan6
	        * pd_dma0_chan7
	          * pd_esai0
	      * pd_dma0_chan0
	        * pd_dma0_chan1
	          [....]
	         * pd_asrc0

We need to make pd_asrc0 dependent on pd_esai0, thus we move
pd_dm0_chan6 node as a child of pd_asrc0.

Thus, the new power domain hierarch will look like this:

	* pd_audio
	  * pd_audio_clk0
	    * pd_audio_clk1
	      * pd_dma0_chan0
	        * pd_dma0_chan1
	          [....]
	         * pd_asrc0
	           * pd_dma0_chan6
	             * pd_dma0_chan7
	              * pd_esai0

Signed-off-by: Daniel Baluta <daniel.baluta@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
(cherry picked from commit 04c4825baad9344e5f3dfa1c69de3957199d3ab0)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qxp-mek-dsp.dts     |   32 ++++++++++++++++++++
 1 file changed, 32 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dsp.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dsp.dts
index ed00618..d975b50 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dsp.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qxp-mek-dsp.dts
@@ -76,6 +76,38 @@
 	status = "okay";
 };
 
+/delete-node/ &pd_dma0_chan6;
+
+&pd_asrc0 {
+	reg = <SC_R_ASRC_0>;
+	power-domains =<&pd_dma0_chan5>;
+	#power-domain-cells = <0>;
+	#address-cells = <1>;
+	#size-cells = <0>;
+
+	pd_dma0_chan6: PD_ESAI_0_RX {
+		reg = <SC_R_DMA_0_CH6>;
+		power-domains =<&pd_asrc0>;
+		#power-domain-cells = <0>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		pd_dma0_chan7: PD_ESAI_0_TX {
+			reg = <SC_R_DMA_0_CH7>;
+			power-domains =<&pd_dma0_chan6>;
+			#power-domain-cells = <0>;
+			#address-cells = <1>;
+			#size-cells = <0>;
+
+			pd_esai0: PD_AUD_ESAI_0 {
+				reg = <SC_R_ESAI_0>;
+				#power-domain-cells = <0>;
+				power-domains =<&pd_dma0_chan7>;
+			};
+		};
+	};
+};
+
 &esai0 {
 	status = "disabled";
 };
-- 
1.7.9.5

